test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -223,6 +230,147 @@ func (data *testData) pingBetweenInterfaces(t *testing.T) error {
+	}
+
+	ovsPorts := strings.Fields(stdout)
+	return ovsPorts, nil
+}
+
+// reconcilationAfterAgentRestart verifies OVS Ports cleanup and IP release.
+func (data *testData) reconcilationAfterAgentRestart(t *testing.T) {
+	ovsPortsBeforeAgentRestart := make(map[string][]string)
+	beforeAgentRestartIPsAndIfaces := make(map[string]map[string]net.IP)
+	for _, pod := range data.pods {
+		ports, err := data.getOVSPortsOnSecondaryBridge(t, pod.nodeName)
+		require.NoError(t, err, "Failed to get Secondary bridge OVS Ports before agent restart")
+		ovsPortsBeforeAgentRestart[pod.nodeName] = ports
+
+		ips, err := data.listPodIPs(pod)
+		require.NoError(t, err, "Failed to get Pod IP before agent restart")
+		beforeAgentRestartIPsAndIfaces[pod.podName] = ips
+	}
+
+	require.NoError(t, data.e2eTestData.RestartAntreaAgentPods(30*time.Second), "Failed to restart Antrea agent Pods")
+
+	ovsPortsAfterAgentRestart := make(map[string][]string)
+	afterAgentRestartIPsAndIfaces := make(map[string]map[string]net.IP)
+	for _, pod := range data.pods {
+		ports, err := data.getOVSPortsOnSecondaryBridge(t, pod.nodeName)
+		require.NoError(t, err, "Failed to get Secondary bridge OVS Ports after agent restart")
+		ovsPortsAfterAgentRestart[pod.nodeName] = ports
+
+		ips, err := data.listPodIPs(pod)
+		require.NoError(t, err, "Failed to get Pod IP after agent restart")
+		afterAgentRestartIPsAndIfaces[pod.podName] = ips
+	}
+
+	// Compare OVS ports before and after restart
+	for nodeName, portsBefore := range ovsPortsBeforeAgentRestart {
+		assert.ElementsMatch(t, portsBefore, ovsPortsAfterAgentRestart[nodeName],
+			"Secondary bridge OVS Ports mismatch after agent restart on node %s", nodeName)
+	}
+
+	// Compare IPs and interfaces before and after restart
+	for podName, ipsBefore := range beforeAgentRestartIPsAndIfaces {
+		assert.Equal(t, ipsBefore, afterAgentRestartIPsAndIfaces[podName],
+			"Pod: %s, Interfaces and IPs mismatch after agent restart", podName)
+	}
+
+	// Remove Pod and check IP released or not.
+	vlanPod := data.pods[1]
+	ifaces := []string{"eth1", "eth2"}
+	ifacesIPs, err := data.listPodIPs(vlanPod)
+	require.NoError(t, err, "Failed to get IPs of Interfaces")
+
+	ovsPortsBeforeDeletion, err := data.getOVSPortsOnSecondaryBridge(t, vlanPod.nodeName)
+	require.NoError(t, err, "Failed to get OVS Ports Before Pod deletion")
+
+	ipPools, err := data.getIPPoolNames(ifaces, vlanPod, ipPoolNamespace)
+	require.NoError(t, err, "Failed to get IPPool")
+
+	require.NoError(t, data.e2eTestData.DeletePodAndWait(defaultTimeout, vlanPod.podName, data.e2eTestData.GetTestNamespace()), "Failed to delete Pod")
+
+	ovsPortsAfterDeletion, err := data.getOVSPortsOnSecondaryBridge(t, vlanPod.nodeName)
+	require.NoError(t, err, "Failed to get OVS Ports After Pod deletion")
+
+	assert.NotEqual(t, ovsPortsBeforeDeletion, ovsPortsAfterDeletion, "OVS Ports for VLAN Pod still exist")
+
+	data.checkIPReleased(ipPools, ifacesIPs, ifaces)
+
```

Comment:
remove

Reply:
Hi @antoninbas , you mean to remove assert and  checkIPReleased function?